elastic/beats
# Disable cleanup_timeout by default in docker and kubernetes autodiscover

## What does this PR do?

Disable `cleanup_timeout` by default in docker and kubernetes autodiscover for all beats except Filebeat.

It is kept to 60 seconds in Filebeat, to give a time to collect logs.

## Why is it important?

Keeping configurations running for some time after containers have stopped is needed in some cases to complete the collection of logs. But in the rest of cases it is not usually needed, and leads to errors when querying endpoints known to be down.
It can also lead to query IPs that are being reused in newer containers, what can be misleading if the newer pod answers because these events will still have the metadata of the old container.

## Checklist


## How to test this PR locally

* Start metricbeat and filebeat in Kubernetes with autodiscover and some configuration.
* Create a pod in Kubernetes that matches the existing configurations.
* Check that metricbeat and filebat start collecting metrics and logs.
* Delete the pod.
* Check that metricbeat stops collecting metrics.
* Check that filebeat stops collecting metrics about 60 seconds later. 

## Related issues

- Fixes #20543.
